pkg/apis/serving/v1alpha2/kfservice_types.go

```diff
@@ -21,36 +21,73 @@ import (
 
 // KFServiceSpec defines the desired state of KFService
 type KFServiceSpec struct {
-	Default ModelSpec `json:"default"`
-	// Canary defines an alternate configuration to route a percentage of traffic.
-	Canary               *ModelSpec `json:"canary,omitempty"`
-	CanaryTrafficPercent int        `json:"canaryTrafficPercent,omitempty"`
+	// Default defines default KFService endpoints
+	// +required
+	Default EndpointSpec `json:"default"`
```

Comment:
Another thought on this I was discussing on Friday. Should this be "PipelineSpec"? Effectively, KFService is an InferencePipeline for a single model. a XYZGraph is an InferenceGraph for multiple models, joining together multiple pipelines.

If we want to start messaging this way in our docs and diagrams, it might help to call this `InferencePipelineSpec`

Reply:
not sure if we can call it pipeline since it is not just inference pipeline, we have `predict`, `explain` endpoints and potentially drift/bias detection in future ?

Reply:
Yeah drift/bias may break the "pipeline" concept.